### PR TITLE
sql : add a test for check whether we could add a same name column wh…

### PR DESCRIPTION
**Pull request by a6802739**
Add a test for #16020. @vivekmenezes

### CODE REVIEW

File: pkg/sql/schema_changer_test.go [hunk 1]
```diff
@@ -1737,6 +1737,64 @@ CREATE TABLE t.test (
 	}
 }
 
+// This test check whether we could add a same column name when this column is in DELETE_ONLY state.
+func TestAddColumnDuringColumnBackfill(t *testing.T) {
```
Comment (vivekmenezes) [inline pkg/sql/schema_changer_test.go]: Can you rename this TestAddColumnDuringColumnDrop()

File: pkg/sql/schema_changer_test.go [hunk 2]
```diff
@@ -1737,6 +1737,64 @@ CREATE TABLE t.test (
...
+	params, _ := createTestServerParams()
+	backfillNotification := make(chan struct{})
+	continueBackfillNotification := make(chan struct{})
+	params.Knobs = base.TestingKnobs{
+		SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
+			RunBeforeBackfill: func() error {
+				if backfillNotification != nil {
+					// Close channel to notify that the schema change has
+					// been queued and the backfill has started.
+					backfillNotification <- struct{}{}
```
Comment (vivekmenezes) [inline pkg/sql/schema_changer_test.go]: No need to send anything through the channel. Just close it

File: pkg/sql/schema_changer_test.go [hunk 3]
```diff
@@ -1737,6 +1737,64 @@ CREATE TABLE t.test (
...
+
+	if err := bulkInsertIntoTable(sqlDB, 1000); err != nil {
+		t.Fatal(err)
+	}
+	// Run the column schema change in a separate goroutine.
+	notification := backfillNotification
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		if _, err := sqlDB.Exec(`ALTER TABLE t.test drop column v;`); err != nil {
```
Comment (vivekmenezes) [inline pkg/sql/schema_changer_test.go]: s/drop column/DROP COLUMN

File: pkg/sql/schema_changer_test.go [hunk 4]
```diff
@@ -1737,6 +1737,64 @@ CREATE TABLE t.test (
 	}
 }
 
+// This test check whether we could add a same column name when this column is in DELETE_ONLY state.
```
Comment (vivekmenezes) [inline pkg/sql/schema_changer_test.go]: // This test checks whether a column can be added using the name of a column that has just been dropped.

**Review by vivekmenezes — COMMENTED** [review]
Can you change the commit message to: 

sql: test that a column recently dropped doesn't recycle its name immediately

related to #16020

Thanks

File: pkg/sql/schema_changer_test.go (hunk 4 above)
Comment (a6802739) [inline pkg/sql/schema_changer_test.go]: Done.

File: pkg/sql/schema_changer_test.go (hunk 1 above)
Comment (a6802739) [inline pkg/sql/schema_changer_test.go]: Done.

File: pkg/sql/schema_changer_test.go (hunk 2 above)
Comment (a6802739) [inline pkg/sql/schema_changer_test.go]: Done. My mistake.

File: pkg/sql/schema_changer_test.go (hunk 3 above)
Comment (a6802739) [inline pkg/sql/schema_changer_test.go]: Done.

**Comment by a6802739** [comment]
@vivekmenezes  PLAT. Thank you.

**Comment by vivekmenezes** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16956#-:-Kp_Tywu9jsyF_yjKT_d:bnfp4nl)*

**Pull request merged**